dist/tools/backport_pr/backport_pr.py

```
@@ -0,0 +1,218 @@
+        if (remote.url.endswith("{}/{}.git".format(ORG, REPO)) or
+                remote.url.endswith("{}/{}".format(ORG, REPO))):
+            return remote
+
+
+def _delete_worktree(repo, workdir):
+    shutil.rmtree(workdir)
+    repo.git.worktree('prune')
+
+
+def main():
+    keyfile = os.path.join(os.environ['HOME'], GITHUBTOKEN_FILE)
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-k", "--keyfile", type=argparse.FileType('r'),
+                        default=keyfile,
+                        help="File containing github token")
+    parser.add_argument("-c", "--comment", action="store_true",
+                        help="Put a comment with a reference under"
+                             "the original PR")
+    parser.add_argument("-n", "--noop", action="store_true",
+                        help="Limited noop mode, creates branch, but doesn't"
+                             "push and create the PR")
+    parser.add_argument("-r", "--release-branch", type=str,
+                        help="Base the backport on this branch, "
+                             "default is the latest")
+    parser.add_argument("--backport-branch-fmt", type=str,
+                        default=BACKPORT_BRANCH,
+                        help="Backport branch format. "
+                             "Fields '{release}' and '{origbranch} will be "
+                             "replaced by the release name and remote branch "
+                             "name.")
+    parser.add_argument('-d', '--gitdir', type=str, default=os.getcwd(),
+                        help="Base git repo to work from")
+    parser.add_argument("PR", type=int, help="Pull request number to backport")
+    args = parser.parse_args()
+
+    gittoken = args.keyfile.read().strip()
+    g = GitHub(token=gittoken)
+    # TODO: exception handling
+    status, user = g.user.get()
+    if status != 200:
+        print("Could not retrieve user: {}".format(user['message']))
+        exit(1)
```

This

```py
status, obj = endpoint.get()
if status != 200:
   print("error")
   exit(val)
```

pattern shows up quite often. Any chance to put this into its own function?

---

That should just raise an exception.

---

The function should raise an exception? Yes. Otherwise, I'm not sure what you mean.

---

Yes that is what I meant indeed.